napari/utils/colormaps/colormap.py

```diff
@@ -522,3 +740,41 @@ def minimum_dtype_for_labels(num_colors: int) -> np.dtype:
     if num_colors <= np.iinfo(np.uint16).max:
         return np.dtype(np.uint16)
     return np.dtype(np.float32)
+
+
+PRIME_NUM_TABLE = [
+    37,
+    61,
+    127,
+    251,
+    509,
+    1021,
+    2039,
+    4093,
+    8191,
+    16381,
+    32749,
+    65521,
+]
```

Comment:
Do you mean to copy thid function from PR and use `lru_cache` and search in longer prime list?

Reply:
Yes, copy the function. With regards to use, there's two options:

- just `_PRIMES = _primes(upto=2**16)`, then use that as the table. (I think just `_PRIMES` is enough for the name.)
- alternatively, don't have the list — instead of bisect, call `_primes(upto=2 * len(label_mapping))[-1]`. This function is run rarely and is extremely fast, so we don't necessarily need to maintain a table of primes manually.